src/core/src/bootstrap/EnvLayer.py

```diff
@@ -79,6 +83,79 @@ def get_package_manager(self):
 
         return ret
 
+    def set_env_var(self, var_name, var_value=None, raise_if_not_success=False):
+        """ Sets an environment variable with var_name and var_value in /etc/environment. If it already exists, it is overwriten. """
+        try:
+            environment_vars = self.file_system.read_with_retry(self.etc_environment_file_path)
+            if environment_vars is None:
+                print("Error occurred while setting environment variable: File not found. [Variable={0}] [Value={1}] [Path={2}]".format(str(var_name), str(var_value), self.etc_environment_file_path))
+                return
+
+            environment_vars_lines = environment_vars.strip().split("\n")
+
+            if var_value is None:
+                # remove environment variable
+                if var_name not in environment_vars:
+                    return
+
+                regex = re.compile('{0}=.+'.format(var_name))
+                search = regex.search(environment_vars)
+                if search is None:
+                    return
+
```

Comment:
Do we need these 2 ifs?
First if finds any part in the file with var_name, could be a comment too while the second one looks for the exact pattern match, which is what we are looking for.

If this is right, then I think just the second if should be okay

Reply:
Yep you're right, the first if would just save the execution of the regex logic. I'll remove the first if and leave it to the pattern matching.